.github/workflows/static-analysis-js.yml

```diff
@@ -39,16 +44,16 @@ jobs:
           node-version: ${{ inputs.NODE_VERSION }}
           registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
 
-      - name: Cache npm dependencies
+      - name: Cache npm/Yarn dependencies
         uses: actions/cache@v3
         with:
-          path: ~/.npm
-          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
+          path: ~/.deps_cache
+          key: ${{ runner.os }}-node-${{ inputs.NODE_VERSION }}-${{ inputs.PACKAGE_MANAGER }}-${{ hashFiles('**/package-lock.json', '**/npm-shrinkwrap.json', '**/yarn.lock') }}
 
-      - name: Install npm dependencies
-        run: npm install
+      - name: Install dependencies
+        run: |
+          ${{ ((inputs.PACKAGE_MANAGER == 'yarn') && 'yarn config set cache-folder ~/.deps_cache') || 'npm config set cache ~/.deps_cache --global' }}
```

Comment:
I know that GitHub Actions still uses Yarn 1 (see e.g. [here](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#:~:text=from%20master%20%3C796213014%3E%29-,Yarn%201.22.19,-Notes%3A)).

However, should this workflow be run on a system where the user manually installed a newer version, `yarn config set cache-folder` won't work because the settings key changed in Yarn >= 2. There, it's called `cacheFolder` ([documentation](https://yarnpkg.com/configuration/yarnrc#cacheFolder)), and the command would therefore be `yarn config set cacheFolder`.